examples/mnist/train_mnist.py

```diff
@@ -85,7 +85,8 @@ def main():
 
     # Take a snapshot for each specified epoch
     frequency = args.epoch if args.frequency == -1 else max(1, args.frequency)
-    trainer.extend(extensions.snapshot(), trigger=(frequency, 'epoch'))
+    trainer.extend(extensions.snapshot(snapshot_on_error=True),
```

Comment:
I think it's better to avoid modifying the example, as this file is often used as a boilerplate and not all users want this behavior. (when an error occurs **during** the 10th iteration, a snapshot named `snapshot_iter_10` will be produced, but users cannot distinguish if it with the snapshot that **completed** the 10th iteration.)